### DISCUSSION TITLE
Screen record without screen capture program

### DISCUSSION THREAD

**Discussion by fzs1933** [Stellarium/stellarium#2151] [Q&A] [answered]
Hi,

I finished my script and tried to record video (OBS) but Stellarium sometimes blinking and I have to start and start again recording until it is flicker-free. Drivers are the latest.
It takes too much time to restart. And restart and restart. And waiting for the faultless run.
Is there any other way to record the running of the script frame by frame (plugin, script, etc.)?
For example: https://beltoforion.de/en/stellarium_video/
But that works - as far as I know - only linux (or I did not understand something).

Thanks
Zsolt

**Comment by alex-w** [comment, marked as answer]
No, Stellarium cannot record video itself (see issue #467). Please try windowed mode to record the screen on Windows.

**Reply by gzotti** [reply]
I just use the Game Bar (Win-G) recording option. It records the mouse cursor, so you will need to have a few seconds of waiting and clip away the start until mouse cursor is off-screen.

**Reply by fzs1933** [reply]
Thanks your answer.
But I was afraid of that...
I read the issue. Windowed mode isn’t good because it ruins the aspect ratio.
I know nothing is good enough for me.
Another question: I linked a Python application. It looks very impressive. Is it true that Georg closed the issue with no need for anything like that, but couldn’t you think about it anyway? :)

**Reply by fzs1933** [reply]
> I just use the Game Bar (Win-G) recording option. It records the mouse cursor, so you will need to have a few seconds of waiting and clip away the start until mouse cursor is off-screen.

Its quality is... :( At least I couldn't get some quality out of it.

**Reply by gzotti** [reply]
I don't know which Python application you linked and what I should have closed here. We know (that headache type of knowing) that scripting will have to change when we move to Qt6, but we may be able to keep ECMAscript. If anybody adds useful Python support this is more than welcome.

**Reply by fzs1933** [reply]
This is:
https://beltoforion.de/en/stellarium_video/
https://github.com/beltoforion/kalstar
If I understand correctly, this could be a solution.

Sorry for my many - sometimes pointless - questions but I have a dream. And there are lot of problems. Most of these are related to programming. Cause I am not programmer, I have to find any kind of solution or find someone who can help. When I realized that the solution goes beyond favor or is too complicated, I look for another way.

Thanks for your patience - again... :)

**Reply by gzotti** [reply]
If this works better for you, go ahead. But it is an external "python wrapper" calling Stellarium with a script created on the fly. It is written for Linux. Maybe it runs on Windows/WSL/Ubuntu with some X11 server with a Linux version of Stellarium. Or you modify the script to run on Windows. Good luck!

**Reply by fzs1933** [reply]
OK. I continue to find the solution.
Thanks

**Comment by shiningsurya** [comment]
hello, sorry to reopen this. 

i think venturing out a better solution would be worth while!
i extensively used stellarium to record videos of astro phenomena for a pub-talk. 

- `kalstar` essentially takes a screenshot and then uses `ffmpeg` to render a video from the frames. i tried it, it is extremely slow. If we want to make 60fps video of say 20seconds, we would need 1200 frames! 
The stellarium itself does not run at 60fps but we sure would want Full HD, 60fps (or better). 
Screenshots are taken by  `core.screenshot` so, it cannot happen in parallel when we are `moveTo` or `zoomTo`. This means, we cannot show camera movements. 

- Using the screen-capture reduces the quality. i used the Windows built-in one but i am not happy with the quality. space is beautiful not like 320p. 
Another issue with screen-capture is you have to be quick when you start/stop recording. You lose control. 

Ideal solution would be something like running without a window and `ffmpeg` exporting video. i think this would involve the following:

1. some way to signal other processes from within Stellarium script to control when to start/stop recording and to pass other arguments (filename, video options). 
2. running Stellarium in memory (like a video stream) and still controlling it. Like a console mode.  Are there any plans to have Stellarium run in interpreter mode? Scripts are run in console, and the "space" is rendered someplace else.

Does anyone have any pointers related to above? i would love to work on this. 
Thanks

**Reply by gzotti** [reply]
There are no plans to implement this, but go ahead if you know what you are doing. 
I have used the Windows capture thingy to record a 2xFullHD (3840x1080, certainly not 320p) video from a script which was good enough for presentation in a public exhibition. Stellarium is no video production engine, so we cannot control whether you achieve 60fps or less.

**Reply by shiningsurya** [reply]
wow, Windows screen capture gave you that high of a resolution (hmm, i wonder why i didn't get good results). 

okay! thanks, i will give it a try this weekend! 
thanks for making stellarium! it's awesome! it's like what `manim` is for math guys, but for astronomers!

**Reply by gzotti** [reply]
You capture your screen resolution, so this was with 2 screens of course. And a Geforce (~1080?) card. Just make sure you disable mouse cursor, as this also gets recorded. Trigger script with keyboard shortcut, after recording trim start/end frames.
